semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -536,6 +541,72 @@ and expr_aux env ?(void = false) e_gen =
        * interpolated strings, but we do not have an use for it yet during
        * semantic analysis, so in the IL we just unwrap the expression. *)
       expr env e
+  | G.Call
+      ( ({ e = N (Id (("module", _), _)); _ } as e),
+        ((_, [ _; G.Arg { e = Record (_, fields, _); _ } ], _) as args) )
+    when is_hcl env.lang -> (
+      (* bTERRAFORM:
+         OK, so here's the idea.
+         Terraform modules are translated in the form of `Call`, but we would like to
+         interpret them as actual function calls to fake function definitions.
+         This step in the translation to IL just manufactures that fake function
+         definition.
+         This function is named the same as the "module name" for a given module,
+         which is of the form `dir/module_name` for a module at directory `dir`.
+         The parameters are all of the `variable` blocks.
+
+         We have to translate into an actual IL Call construct with the correct
+         name so that we can let the taint engine do the work.
+      *)
+      let mod_sources, _, mod_args =
+        Common.partition_either3
+          (function
+            | G.F
+                {
+                  s =
+                    DefStmt
+                      ( {
+                          name =
+                            EN
+                              (Id
+                                 ( ("source", _),
+                                   {
+                                     id_resolved =
+                                       {
+                                         contents = Some (G.ResolvedName _, _);
+                                         _;
+                                       };
+                                     _;
+                                   } ) as name);
+                          _;
+                        },
+                        VarDef _ );
+                  _;
+                } ->
+                Left3 (G.N name |> G.e)
+            | G.F
+                {
+                  s =
+                    DefStmt
+                      ( { name = EN (Id (arg_id, _)); _ },
+                        VarDef { vinit = Some arg_exp; _ } );
+                  _;
+                } ->
+                Right3 (G.ArgKwd (arg_id, arg_exp))
+            | _ ->
+                (* bTODO: Ignore all other cases. They correspond neither to the module's name, or its arguments. *)
```

Comment:
What are these other cases (for example)?

Reply:
I will be more explicit next time. But I expect that you answer that question in a code comment, e.g. adding a link to the Terraform docs where those "other cases" are specified or something.